### DISCUSSION TITLE
server answer in case of illegitimate requests

### DISCUSSION THREAD

**Discussion by boss1819** [apernet/hysteria#724] [Q&A] [answered]
How does hysteria server respond if a client with wrong password (or malformed packet) makes a request?

How does that change in case of using obfuscation?

thanks

**Comment by tobyxdd** [comment]
This is pretty clearly explained in the protocol specs. If authentication fails, the server will act as a normal HTTP/3 server and forward the request to whatever upstream you set in masquerade, or return 404 if you don't have one.

**Reply by boss1819** [reply]
thanks, I think it is clear in case no obfuscation used.
as far as I understood from documentation, in case obfuscation used on server side, and client uses wrong obfuscation password, connection will time out and no reply from server will be given. basically the server does not act as http3 server anymore

**Reply by tobyxdd** [reply, marked as answer]
Yes. Obfuscation is essentially a layer of packet encryption. Wrong password = can't decrypt = malformed packet = no response
